BucketSort.c

```diff
@@ -0,0 +1,181 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "BucketSort.h"
+#include "QuickSort.h"
+
+#define INFTY 1.0e+6
+#define TOL 1.0e-14
+#define mabs(a) ((a) >= 0 ? (a) : -(a))
+#define sgn(a) ((a) == 0 ? 0 : ((a) > 0  ? 1 : -1 ))
+#define max(a,b) ((a) >= (b) ? (a) : (b))
+#define min(a,b) ((a) <= (b) ? (a) : (b))
+
+void dial_list_init(struct mylist *list,int ind);
+void dial_bucket_init(struct mybucket *bucket,int iskip,double gap);
+void print_buckets(int Nbuckets,struct mybucket *bucket);
+int adjust_bucket(int ind,double newval,double g,int Nbuckets,struct mybucket *bucket,struct mylist *list);
+int find_bucket(double utemp,double g);
+void myfree(struct bucket_sort_stuff  *BB);
+void start_filling_buckets(struct bucket_sort_stuff  *BB,int Nbuckets,struct mybucket *bucket,
+		struct mylist *list,double gap,int *bdry,double *blist,int bcount);
+int find_number_of_buckets(double gap,double maxgap);
+void form_list_of_new_valid_points(struct mybucket *bucket,int *newlist,int *empty_count);
+//---------------------------------------------------------------
```

Comment:
Please remove this kind of "//------" comment.

Reply:
If you do want these separators gone I'll remove them but later. These separators help me find functions in the library files faster, and I am still editing them.

Reply:
OK, no problem.